src/main/java/org/jabref/gui/fieldeditors/EditorTextArea.java

```diff
@@ -49,14 +50,16 @@ public EditorTextArea(String text) {
     }
 
     /**
-     * Adds the given list of menu items to the context menu.
+     * Adds the given list of menu items to the context menu. The usage of {@link Supplier} prevents that the menus need
+     * to be instantiated at this point. They are populated when the user needs them which prevents many unnecessary
+     * allocations when the main table is just scrolled with the entry editor open.
      */
-    public void addToContextMenu(List<MenuItem> items) {
+    public void addToContextMenu(Supplier<List<MenuItem>> items) {
         TextAreaSkin customContextSkin = new TextAreaSkin(this) {
             @Override
             public void populateContextMenu(ContextMenu contextMenu) {
                 super.populateContextMenu(contextMenu);
-                contextMenu.getItems().addAll(0, items);
+                contextMenu.getItems().addAll(0, items.get());
```

Comment:
If I understand everything correctly, we could build the supplier right here and still accept a `List<MenuItem>` as method parameter. That would keep the changes very localized.

More discussion see below.

Reply:
Hmm, then I misunderstood the whole thing. Every text-field editor needs to attach its own context menu. Als long as I have wrapped it in a supplier, the code is not instantiated. So how is for instance `IdentifierEditor` supposed to attach a non-instantiated context menu?

```
textArea.addToContextMenu(EditorMenus.getDefaultMenu(textArea));
```

Assume we don't do the wrapping inside `EditorMenus`, then the menu would be instantiated exactly there. Except when I put the `Supplier` wrapper at this place. But then we have the suppliers scattered over even more classes. What do I miss?

Reply:
No, Jörg, the supplier has to be built outside of this method since otherwise the context menu items are already created. With the help of a supplier, we are able to delay the creation until the user opens the context menu (since only then the `get` method is called).

Reply:
Ok, then I was wrong. Thanks for clarifying! In that case, please disregard my comment and keep the solution.